README.md

```diff
@@ -87,6 +87,64 @@ Start the containers using docker compose:
 docker-compose up
 ```
 
+### Testing
+
+#### Running the standard test-suite
+
+running locally, in the `osmaxx-py` directory (using a virtualenv):
+
+```shell
+DJANGO_SETTINGS_MODULE=config.settings.test ./manage.py test
+```
+
+*Notice*: Since the ampq Service is not available, you'll get a 4 failing tests. Therefore it is recommended to
+run them in a docker container, as described below.
+
+Using docker and docker-compose:
+ 
+```shell
+docker-compose run --rm webappdev bash -c 'DJANGO_SETTINGS_MODULE=config.settings.test ./manage.py test'
+```
+
+#### Running the test-suite and integration tests
+
+There's a script to facilitate testing including some docker integration tests:
+
+```shell
+./test.sh
+```
+
+#### Running end to end tests (e2e)
+
+**Warnings**: 
+
+* These run for quite a long time (over five Minutes!). 
+* The containers are being destroyed at the beginning (creating a clean state). Resulting in
+    * Only run one test at a time
+    * don't run on production!
+    * don't run docker-compose in parallel
+    * exiting data is being destroyed 
+
+_Prerequisites_: an activated virtualenv!
```

Comment:
Should be a subsection title. If nesting becomes too deep for that, consider a separate document.